### ISSUE TITLE
icon request: desk-light

### ISSUE THREAD

**Issue by hvbhome** [Templarian/MaterialDesign#3057]
as title

**Comment by CoDEmanX** [comment]
Maybe make the light bulb a bit larger somehow?

**Comment by Tarilonte** [comment]
I agree with you. And also maybe it's better to have a outlined bulb.

![image](https://user-images.githubusercontent.com/29761068/39024822-a41836fc-4419-11e8-9a7e-5274cd19d2e0.png)
[desk-light_2.zip](https://github.com/Templarian/MaterialDesign/files/1930194/desk-light_2.zip)

**Comment by hvbhome** [comment]
nice!
you might even consider taking the bulb out completely?
also, create a desk-light-off while you're at it with a slash?
thanks!
(a little flip-switch at the top of the light or the foot could be a nice tweak, not sure if this is an option because of the pixel width though..)

**Comment by CoDEmanX** [comment]
The lightshade looks too small wthout the bulb IMO:

![grafik](https://user-images.githubusercontent.com/1309249/39069943-c7053322-44e1-11e8-8751-ba24394c7abd.png)

A pronounced light bulb looks really odd:

![grafik](https://user-images.githubusercontent.com/1309249/39070389-852c74b8-44e3-11e8-9ef6-27978efe73f7.png)

So many simply a larger lampshade?

A switch would most likely be unrecognizable at normal size.

**Comment by Tarilonte** [comment]
New version with On/Off states

![image](https://user-images.githubusercontent.com/29761068/39077926-e2c6ad12-44db-11e8-9871-68f0bb4a3acc.png) ![image](https://user-images.githubusercontent.com/29761068/39077920-d8cf3e28-44db-11e8-9c32-9a3b595f154f.png)

**Comment by GreenTurtwig** [comment]
Here's a design that's more consistent with out other light icons:
[![](https://user-images.githubusercontent.com/15905797/39086718-44ab17f4-458e-11e8-8c5f-fb1dae331151.png)](https://github.com/Templarian/MaterialDesign/files/1934791/lamp-desk.zip)

@hvbhome We won't be including any "off" variants as the UI for a light usually contains a toggle switch. Also we would have create a variant for every light design we currently have, some of which won't be possible or easily done.

**Comment by hvbhome** [comment]
so happy with this a big thanks !
I hope you don't mind me asking again about this....but...
could we have this light mirrored left-right, so we have desk-lamp-right and desk-lamp-left...
seems silly detailed but would be so hyper convenient.
cheers!

**Comment by PeterShaggyNoble** [comment]
@hvbhome: You can easily flip it yourself.

**Comment by hvbhome** [comment]
> You can easily flip it yourself.

don't want to seem a complete idiot, but how would i do that using the `icon: mdi:desk-light` setting in Home-assistant?

**Comment by GreenTurtwig** [comment]
You'd have to contribute the feature to Home Assistant yourself or submit a request. Home Assistant simply uses our set.

**Comment by hvbhome** [comment]
> or submit a request

thought I'd done that ..;-) Would have been best if implemented in your set, and used by Home-assistant. No further manual tweaking necessary.
Maybe there's a way to easily do the flipping in HA, haven't seen that before, will certainly check.
thx,
Marius

**Comment by PeterShaggyNoble** [comment]
@hvbhome, how do you apply styles such as `fill`s to icons in HA? Could you not `transform` the icon there?

**Comment by hvbhome** [comment]
> how do you apply styles such as fills to icons in HA? Could you not transform the icon there?

tbh I wouldn't know, but I dont think thats what I asked is it? What I meant to ask was if you guys wouldn't;t best do the icon-styling, to have Home-assistant import them. I wouldn't have suggested doing things like that myself....
transform is, as far as I know now, not an option within the possibilities of regular home-assistant yaml or customizing by templating

**Comment by goyney** [comment]
Home Assistant uses `ha-state-icon[data-domain=light][data-state=on]` as a CSS selector to recolor an icon to an "on" state. From what I can see in my own HASS control panel, there is no identifiable information in the DOM to target a specific domain, entity, or switch.

I would recommend opening a feature request with Home Assistant to allow custom orientations with their icons. Perhaps you can suggestion them modifying the YAML schema to allow something like:

```yaml
mdi:lightbulb:flip-h
mdi:lightbulb:rotate-45
```

Or suggest better identifiers in the DOM that can be targeted with CSS via their custom theme feature.

**Comment by hvbhome** [comment]
thanks, posted the question in Home-assistant community:

https://community.home-assistant.io/t/customui-discussion-thread/48694/190?u=mariusthvdb
